ctapipe/core/feature_generator.py

```diff
@@ -0,0 +1,30 @@
+from .component import Component
+from .expression_engine import ALLOWED_GLOBALS, ExpressionEngine
+from .traits import List
+
+
+class FeatureGeneratorException(TypeError):
+    """Signal a problem with a user-defined selection criteria function"""
+
+
+class FeatureGenerator(Component):
+    features = List().tag(config=True)
```

Comment:
I think this should be a dict mapping feature names to expressions.

Reply:
Yes, that would fit better. What do you think of using a dict also in the QualityQuery? Now that we removed the possibility to use `lambda` in the expressions, we can also change the type.

Reply:
We might depend on order the order of the checks when merging and I am not sure everything we use guarantees dict order (python does, but e.g. json and toml don't, so their python parsers also might not).

Reply:
But this is also the case for here, right? Generating a feature from another generated feature

Reply:
That would be solvable  though by checking the dependencies, which would be a  lot nicer than requiring the user to make it in order anyways.

Reply:
It was a list originally because the order does in fact matter here, at least if you want useful statistics you want to be able to specify the order in which to apply the quality checks.